web/yo/app/scripts/controllers/gene.js

```diff
@@ -3310,14 +3310,24 @@ angular.module('oncokbApp')
         $scope.treatmentRef = data.treatmentRef;
         var geneName = data.geneName;
 
-        $scope.saveCallback = function(newTreatmentName, oldContent){
-            $scope.treatmentRef.name = newTreatmentName;
+        $scope.saveTherapiesCallback = function(newTreatmentName, oldContent){
             var name_uuid = $scope.treatmentRef.name_uuid;
-            mainUtils.setUUIDInReview(name_uuid);
-            var therapyObject = {
-                'name': newTreatmentName,
-                'status': 'latest'
-            };
+            if(!_.isEmpty($scope.treatmentRef.name)) {
+                if (_.isUndefined($scope.treatmentRef.name_review)) {
```

Comment:
Hi @Yizhuzhu, I think you can merge these if statements together (line 3316 ~ 3324).
```
if (_.isUndefined($scope.treatmentRef.name_review)) {
       $scope.treatmentRef.name_review = {
                'updatedBy' : $rootScope.me.name,
                'updateTime':  new Date().getTime(),
       };
} else if (_.isUndefined($scope.treatmentRef.name_review.updatedBy) || _.isUndefined($scope.treatmentRef.name_review.updateTime)) {
       $scope.treatmentRef.name_review.updatedBy = $rootScope.me.name;
       $scope.treatmentRef.name_review.updateTime = new Date().getTime();
}

```